lxd/firewall/interfaces.go

```diff
@@ -0,0 +1,32 @@
+package firewall
+
+import (
+	"net"
+
+	"github.com/lxc/lxd/lxd/device"
+	deviceConfig "github.com/lxc/lxd/lxd/device/config"
+)
+
+
+// Firewall represents an LXD firewall.
+type Firewall interface {
+	// Lower-level Functions
+	NetworkClear(name string, protocol string, table string) error
+	InstanceClear(inst device.Instance, protocol string, table string) error
+	VerifyIPv6Module() error
+
+	// Proxy Functions
+	InstanceProxySetupNAT(protocol string, ipAddr net.IP, comment string, connType, address, port string, cPort string) error
+
+	// NIC Bridged Functions
+	InstanceNicBridgedRemoveFilters(m deviceConfig.Device, ipv4 net.IP, ipv6 net.IP) error
+	InstanceNicBridgedSetFilters(m deviceConfig.Device, ipv4 net.IP, ipv6 net.IP, name string) error
+
+	// Network Functions
+	NetworkSetupAllowForwarding(protocol string, name string, should_accept bool) error
+	NetworkSetupNAT(protocol string, name string, is_after bool, args ...string) error
```

Comment:
What's `is_after` meant to do?
Go also dislikes underscores in variable/argument names instead preferring camel case, so `isAfter` in this case, though a more descriptive name may be useful.

Reply:
Would `shouldPrepend` be better? The purpose of the variable is to allow the same function to be used for both appending and prepending rules. Another option would be to split it into two functions, perhaps `NetworkSetupNATAppend` and `NetworkSetupNATPrepend`?

Reply:
Ok, I'd probably just call it `prepend` then and reverse the logic to match the new name.